zipline/finance/performance/tracker.py

```diff
@@ -459,14 +486,82 @@ def handle_simulation_end(self):
             index=self.cumulative_risk_metrics.cont_index,
             data=self.cumulative_risk_metrics.algorithm_returns_cont)
         acl = self.cumulative_risk_metrics.algorithm_cumulative_leverages
+        cvar = self._calculate_rolling_expected_shortfall(data_portal)
 
         risk_report = risk.RiskReport(
-            ars,
-            self.sim_params,
+            algorithm_returns=ars,
+            expected_shortfalls=cvar,
+            sim_params=self.sim_params,
             benchmark_returns=bms,
             algorithm_leverages=acl,
             trading_calendar=self.trading_calendar,
             treasury_curves=self.treasury_curves,
         )
 
         return risk_report.to_dict()
+
+    def _calculate_rolling_expected_shortfall(self, data_portal):
+        sim_params = self.sim_params
+        lookback_days = zp.DEFAULT_CVAR_LOOKBACK_DAYS
+
+        # Create a data frame of asset weights on each day of the simulation.
+        # If an asset was not held on a given date, it is assigned a weight of
+        # zero.
+        weights = pd.DataFrame(
+            self.position_weights.tolist(),
+            index=sim_params.sessions.normalize(),
+        ).fillna(0)
+        assets = weights.columns.tolist()
+
+        # If we are near the start date of our data, just use the data
+        # available. Otherwise, use the full default number of lookback days.
+        days_before_start = min(
+            self.trading_calendar.session_distance(
+                data_portal.first_day_of_data, sim_params.start_session,
+            ),
+            lookback_days,
+        )
+
+        asset_returns = zp.asset_returns_for_cvar(
+            assets=assets,
+            benchmark=self.benchmark_asset,
+            data_portal=data_portal,
+            end_date=sim_params.end_session,
+            lookback_days=len(sim_params.sessions) + days_before_start,
+        )
+
+        def cvar_of_df(df):
```

Comment:
I prototyped an alternative to this calculation using just numpy here: https://github.com/quantopian/zipline/pull/1874/commits/e8fe43db58c09f1b25377d4c2ecf27992ebb3108.